tiledb/sm/cpp_api/current_domain.h

```diff
@@ -0,0 +1,178 @@
+  /* ********************************* */
+
+  /**
+   * Creates a TileDB current_domain object.
+   *
+   * @param ctx TileDB context
+   * @param type The TileDB currentDomain type
+   */
+  explicit CurrentDomain(const Context& ctx)
+      : ctx_(ctx) {
+    tiledb_current_domain_t* cd;
+    ctx.handle_error(tiledb_current_domain_create(ctx.ptr().get(), &cd));
+    current_domain_ = std::shared_ptr<tiledb_current_domain_t>(cd, deleter_);
+  }
+
+  /**
+   * Creates a TileDB current_domain object.
+   *
+   * @param ctx TileDB context
+   * @param cd The TileDB currentDomain C api pointer
+   */
+  CurrentDomain(const tiledb::Context& ctx, tiledb_current_domain_t* cd)
+      : ctx_(ctx) {
+    current_domain_ = std::shared_ptr<tiledb_current_domain_t>(cd, deleter_);
+  }
+
+  CurrentDomain(const CurrentDomain&) = default;
+  CurrentDomain(CurrentDomain&&) = default;
+  CurrentDomain& operator=(const CurrentDomain&) = default;
+  CurrentDomain& operator=(CurrentDomain&&) = default;
+
+  /* ********************************* */
+  /*                API                */
+  /* ********************************* */
+
+  /**
+   * Returns a shared pointer to the C TileDB currentDomain object.
+   *
+   * @return The C API pointer
+   */
+  std::shared_ptr<tiledb_current_domain_t> ptr() const {
+    return current_domain_;
+  }
+
+  /**
+   * Returns the currentDomain type.
+   *
+   * @return The currentDomain type.
+   */
+  tiledb_current_domain_type_t type() const {
+    tiledb_current_domain_type_t type;
+    auto& ctx = ctx_.get();
+    ctx.handle_error(
+        tiledb_current_domain_get_type(current_domain_.get(), &type));
+    return type;
+  }
+
+  /**
+   * Set a N-dimensional rectangle representation on a current domain.
+   * Error if the current domain passed is not empty.
+   *
+   * @param ndrect The N-dimensional rectangle to be used.
+   * @return Reference to this `currentDomain` instance.
+   */
+  CurrentDomain& set_ndrectangle(const NDRectangle& ndrect) {
+    auto& ctx = ctx_.get();
+    ctx.handle_error(tiledb_current_domain_set_ndrectangle(
+        current_domain_.get(), ndrect.ptr().get()));
+
+    return *this;
+  }
+
+  /**
+   * Get the N-dimensional rectangle associated with the current domain object,
+   * error if the current domain is empty or a different representation is set.
+   *
+   * @return The N-dimensional rectangle of the current domain
+   */
+  NDRectangle ndrectangle() const {
+    tiledb_ndrectangle_t* nd;
+    auto& ctx = ctx_.get();
+    ctx.handle_error(
+        tiledb_current_domain_get_ndrectangle(current_domain_.get(), &nd));
+    return NDRectangle(ctx, nd);
+  }
+
+  /**
+   * Check if the current_domain is empty
+   *
+   * @return True if the currentDomain is empty
+   */
+  bool is_empty() const {
+    uint32_t ret;
+    auto& ctx = ctx_.get();
+    ctx.handle_error(
+        tiledb_current_domain_get_is_empty(current_domain_.get(), &ret));
+    return static_cast<bool>(ret);
```

Comment:
```suggestion
    return ret != 0;
```

Reply:
They both do exactly the same, I believe casting here is more clear and a better C++ practice from what I read